modules/navigation/src/main/java/com/sygic/maps/module/navigation/NavigationFragment.kt

```diff
@@ -170,16 +197,21 @@ class NavigationFragment : MapFragmentWrapper<NavigationFragmentViewModel>() {
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
 
-        fragmentViewModel = viewModelOf(NavigationFragmentViewModel::class.java, arguments)
+        fragmentViewModel = viewModelOf(NavigationFragmentViewModel::class.java, arguments).apply {
+            this.activityFinishObservable.observe(
```

Comment:
ma to mat hardcodnuty nazov `activityFinishObservable`? nie je to konfigurovatelne? To by malo byt skor nieco ako `rightButtonAction.observe`, nie?

Reply:
oki, upravím. díky

Reply:
na druhou stranu to není Infobar VM, ale přímo NavigationFragmentViewModel, tak ten by to i takto mohl mít, však je to jeho vlastní VM, ale i tak to přejmenuji.

Reply:
VM rozhodne jaké observable se zavolá, fragment se na něj musí jen slepě subscribnout. pokud by tam byl další button, který by dělal to stejné, také se to nebude řešit tu, ale ve VM. Vrátím to nakonec zpět, v další branch, který z této vychází to původní dává větší smysl.

Reply:
jj chapem, co pises, stale mi to ale nedava zmysel. musim si to pozriet asi cele, ako sa konfiguruju tie buttony, ale cakal by som, ze aj tento "default" bude pouzivat rovnaky princip, ako ked sa napajas z app a overridujes ten button. Preto mi pride divne, ze tu niekde bude observable na finish, ktore sa v pripade overridu ani nepouzije nikdy...